bookmarks/views.py

```diff
@@ -1,25 +1,37 @@
 from django.shortcuts import render, redirect
 from django.contrib import messages
 from bookmarks.models import Bookmark
+from grades.models import Course
 from bookmarks.forms import BookmarkForm
 
 
 def Create_Bookmarks(request):
     if request.method == "POST":
         form = BookmarkForm(request.POST)
-        user = request.user
+        courses = form.fields["course"].queryset = Course.objects.filter(
+            user=request.user)
         if form.is_valid():
             url = form.cleaned_data['url']
             urlname = form.cleaned_data['urlname']
             course = form.cleaned_data['course']
-            bookmark = Bookmark(user=user, course=course, url=url, urlname=urlname)
+            bookmark = Bookmark(course=course, url=url, urlname=urlname)
             bookmark.save()
-            messages.success(request, "Bookmark saved successfully")
-            return redirect('Bookmarks')
+            messages.success(request, "Bookmark saved successfully!")
+            return redirect('bookmarks:Create_Bookmarks')
     else:
         form = BookmarkForm()
-    bookmarks = Bookmark.objects.filter(user=request.user)
+    courses = Course.objects.filter(user=request.user)
+    bookmarks = Bookmark.objects.all()
     return render(request, 'bookmarks/bookmarks.html', {
         'bookmarks': bookmarks,
+        'courses': courses,
         'form': form,
     })
+
+
+def Remove_bookmarks(request, pk):
+    bm = Bookmark.objects.get(pk=pk)
+    if request.method == 'POST':
+        bm.delete()
```

Comment:
you have a classmethod remove_bookmark on the Bookmarks model. Why aren't you using it?